ansible/roles/test/files/ptftests/inner_hash_test.py

```diff
@@ -0,0 +1,505 @@
+                dport = random.randint(0, 65535) if hash_key == 'dst-port' else 80
+                ip_proto = self._get_ip_proto() if hash_key == 'ip-proto' else 6
+
+                (packet_port_index, _) = self.check_ip_route(hash_key, outer_encap_format, src_port, ip_src, ip_dst, sport, dport, ip_proto)
+                if self.symmetric_hashing and hash_key != 'ip-proto':
+                    # Send the same packet with reversed tuples and validate that it lands on the same port
+                    rand_src_port = int(random.choice(self.src_ports))
+                    (rPacket_port_index, _) = self.check_ip_route(hash_key, outer_encap_format, src_port, ip_dst, ip_src, dport, sport, ip_proto)
+                    assert packet_port_index == rPacket_port_index
+
+                hit_count_map[packet_port_index] = hit_count_map.get(packet_port_index, 0) + 1
+            logging.info("outer_encap_fmts={}, hash_key={}, hit count map: {}".format(outer_encap_format, hash_key, hit_count_map))
+            if hash_key == 'outer-tuples':
+                self.check_all_packets_hash_to_same_nh(self.next_hop.get_next_hop(), hit_count_map)
+            else:
+                self.check_balancing(self.next_hop.get_next_hop(), hit_count_map)
+
+
+    def check_ip_route(self, hash_key, outer_encap_format, src_port, ip_src, ip_dst, sport, dport, ip_proto):
+        if (outer_encap_format == 'vxlan'):
+            (matched_index, received) = self.check_ip_route_vxlan(hash_key, src_port, ip_dst, ip_src, dport, sport, ip_proto)
+        elif (outer_encap_format == 'nvgre'):
+            (matched_index, received) = self.check_ip_route_nvgre(hash_key, src_port, ip_dst, ip_src, dport, sport, ip_proto)
+        else:
+            raise Exception("Logic issue, unexpected outer encap format {}".format(outer_encap_format))
+
+        matched_port = self.exp_port_list[matched_index]
+        return (matched_port, received)
+
+
+    def _get_ip_proto(self, ipv6=False):
+        # ip_proto 2 is IGMP, should not be forwarded by router
+        # ip_proto 254 is experimental
+        # MLNX ASIC can't forward ip_proto 254, BRCM is OK, skip for all for simplicity
+        skip_ports = [2, 254]
+        if ipv6:
+            # Skip ip_proto 0 for IPv6
+            skip_ports.append(0)
+
+        while True:
+            ip_proto = random.randint(0, 255)
+            if ip_proto not in skip_ports:
+                return ip_proto
+
+
+    def generate_inner_pkt(self, sport, dport, ip_src, ip_dst, ip_proto):
+        rand_int = random.randint(1, 99)
+        src_mac = '00:12:ab:34:cd:' + str(rand_int)
+        dst_mac = str(rand_int) + ':12:ab:34:cd:00'
+        pkt = simple_tcp_packet(
+                        eth_dst=dst_mac,
+                        eth_src=src_mac,
+                        ip_src=ip_src,
+                        ip_dst=ip_dst,
+                        tcp_sport=sport,
+                        tcp_dport=dport,
+                        ip_ttl=64)
```

Comment:
@anish-n if we don't want to use a default value, can we have a named definition here?

Reply:
Same comment on hard coded value as being potentially acceptable for test purposes,